hw_5.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+"""HW_5.ipynb
+
+## Задание 1
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %pip install seaborn
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+def read_gff(file):
+    return pd.read_csv(f'{file}', 
+                       sep = '\t',
+                       comment='#',
+                       names=['chromosome', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes'])
+
+def read_bed6(file):
+    return pd.read_csv(f'{file}', 
+                       sep='\t',
+                       names=['chromosome', 'start', 'end','name', 'score', 'strand'])
+
+
+rrna_annotation = read_gff('data/rrna_annotation.gff')
+rrna_annotation['attributes'] = rrna_annotation['attributes'] \
+    .replace(['Name=16S_rRNA.*$', 'Name=23S_rRNA.*$','Name=5S_rRNA.*$'], 
+             ['16S', '23S', '5S'],
+             regex=True)
+alignment = read_bed6('data/alignment.bed')
+
+rna_23s = rrna_annotation[rrna_annotation['attributes'] == '23S']['chromosome'].value_counts().rename('23S')
+rna_16s = rrna_annotation[rrna_annotation['attributes'] == '16S']['chromosome'].value_counts().rename('16S')
+rna_5s = rrna_annotation[rrna_annotation['attributes'] == '5S']['chromosome'].value_counts().rename('5S')
+
+rna_count_table = pd.concat([rna_5s, rna_16s, rna_23s], axis = 1) \
+    .sort_index() \
+    .reset_index(level=0) \
+    .rename({'index' : 'Sequence'}, axis = 'columns')
+
+plt.figure(0)
+rna_count_table.plot(x='Sequence', y=['16S', '23S', '5S'], kind='bar', figsize=(20,10))
+plt.ylabel('Count')
+plt.legend(title='RNA type')
+plt.show()
+
+rrna_annotation['annotation_interval'] = rrna_annotation.apply(lambda row: (row.start, row.end), axis = 1)
+alignment['alignment_interval'] = alignment.apply(lambda row: (row.start - 1, row.end + 1), axis = 1)
+merged = rrna_annotation.merge(alignment, on='chromosome')
+merged = merged[merged.apply(lambda row : 
+                                 row.alignment_interval[0] >= row.annotation_interval[0] and 
+                                 row.alignment_interval[1] <= row.annotation_interval[1],
+                             axis = 1)]
```

Comment:
Сравнение должно быть наоборот, по сути ты здесь проверяшь, что контиг попадает в интервал рРНК, а нужно наоборот. Также это можно немного упростить
```suggestion
merged = merged.query("start_x >= start_y and end_x <= end_y")
```